backend/catalogue/management/commands/generate_layer_previews.py

```diff
@@ -1,103 +1,188 @@
+
 import geopandas
 import geoplot
-import matplotlib.pyplot as plt
-import base64
 import matplotlib.image as mpimg
-import functools
+import matplotlib.pyplot as plt
+import numpy
+import rasterio
+from rasterio import transform, warp
+from pyproj import CRS, Transformer
+from catalogue.emails import email_generate_layer_preview_summary
+from catalogue.models import Layer
+from django.core.files.storage import default_storage
+from django.core.management.base import BaseCommand
 from matplotlib.image import BboxImage
 from matplotlib.transforms import Bbox, TransformedBbox
-from shapely.geometry import shape, LineString, MultiLineString, Polygon, MultiPolygon
-from catalogue.emails import email_generate_layer_preview_summary
+from PIL import Image, UnidentifiedImageError
+from shapely.geometry import (LineString, MultiLineString, MultiPolygon,
+                              Polygon, shape)
 
-from catalogue.models import Layer
+# pylint: disable=line-too-long
+# pylint: disable=missing-function-docstring
+# pylint: disable=missing-class-docstring
+# pylint: disable=redefined-outer-name
 
 Image.MAX_IMAGE_PIXELS = None
 
 # We not able to simply *add* the basemap image file to our GitHub tracking, as it's too large.
 # For now, the image will be added to deployments manually.
-basemap = Image.open(default_storage.open('land_shallow_topo_21600.tif'))
-
-def bounds_to_image_info(bounds):
-    x_delta = (bounds['east'] - bounds['west'] + 360) % 360
-    x_delta = x_delta if x_delta != 0 else 360
-    y_delta = bounds['north'] - bounds['south']
+BASEMAP_FILEPATH = default_storage.path('land_shallow_topo_21600.tif')
+BASEMAP_CRS = 'EPSG:4326'
+BASEMAP_BOUNDS = (-180, -90, 180, 90)
+DST_WIDTH = 386
+
+def get_image_details_from_projected_layer_bounds(min_x: float, min_y: float, max_x: float, max_y: float, crs: str) -> tuple:
+    """
+    Get image details from the projected layer bounds.
+
+    Args:
+        min_x (float): Minimum x coordinate
+        min_y (float): Minimum y coordinate
+        max_x (float): Maximum x coordinate
+        max_y (float): Maximum y coordinate
+        crs (str): The coordinate reference system (e.g. 'EPSG:4326')
+
+    Returns:
+        tuple: (width, height, x_delta, y_delta, aspect_ratio)
+    """
+    crs_obj = CRS.from_string(crs)
+    x_delta = max_x - min_x
+    # Handle cases where the CRS wraps around (e.g. EPSG:4326) and the bounding box crosses the antimeridian
+    if crs_obj.is_geographic and x_delta < 0:
+        x_delta += crs_obj.area_of_use.east - crs_obj.area_of_use.west
+    y_delta = max_y - min_y
     aspect_ratio = x_delta / y_delta
-    width = 386
-    height = round(width / aspect_ratio)
+    height = round(DST_WIDTH / aspect_ratio)
+    return ( DST_WIDTH, height, x_delta, y_delta, aspect_ratio )
 
-    return {
-        'width': width,
-        'height': height,
-        'x_delta': x_delta,
-        'y_delta': y_delta,
-        'aspect_ratio': aspect_ratio
-    }
```

Comment:
I'm curious why you explicitly changed this from a dict to a tuple?  (I get nervous when I see large tuples like this; especially when the values are a bit of a grab-bag.  (minx,miny,maxx,maxy) is at least guessable, but here you discarded explicit names in favour of... convenience, I'm guessing?  Granted python's lack of dict-destructuring is annoying, but it's a small trade-off I'd usually argue.)

Reply:
I had another look at it and found that it doesn't really make sense to have all the information be calculated in a single function. Not every place uses `width`/`height`, and not every place uses `x_delta`/`y_delta`.

I've separated it into two functions for getting the coordinate deltas and image dimensions, and for `aspect_ratio` I've made it so that's calculated on the spot.